apps/api/v2/src/ee/schedules/schedules.repository.ts

```diff
@@ -0,0 +1,133 @@
+import { CreateScheduleInput } from "@/ee/schedules/inputs/create-schedule.input";
+import { UpdateScheduleInput } from "@/ee/schedules/inputs/update-schedule.input";
+import { CreateAvailabilityInput } from "@/modules/availabilities/inputs/create-availability.input";
+import { PrismaReadService } from "@/modules/prisma/prisma-read.service";
+import { PrismaWriteService } from "@/modules/prisma/prisma-write.service";
+import { BadRequestException, Injectable } from "@nestjs/common";
+import { Prisma } from "@prisma/client";
+
+@Injectable()
+export class SchedulesRepository {
+  constructor(private readonly dbRead: PrismaReadService, private readonly dbWrite: PrismaWriteService) {}
+
+  async createScheduleWithAvailabilities(
+    userId: number,
+    schedule: CreateScheduleInput,
+    availabilities: CreateAvailabilityInput[]
+  ) {
+    const createScheduleData: Prisma.ScheduleCreateInput = {
+      user: {
+        connect: {
+          id: userId,
+        },
+      },
+      name: schedule.name,
+      timeZone: schedule.timeZone,
+    };
+
+    if (availabilities.length > 0) {
+      createScheduleData.availability = {
+        createMany: {
+          data: availabilities.map((availability) => {
+            return {
+              days: availability.days,
+              startTime: availability.startTime,
+              endTime: availability.endTime,
+              userId,
+            };
+          }),
+        },
+      };
+    }
+
+    const createdSchedule = await this.dbWrite.prisma.schedule.create({
```

Comment:
Could you return the Promise here rather than awaiting, and having the function call be the one awaiting?

Reply:
What is the benefit of doing that?

As far as I understand there is no need - "this.dbWrite.prisma.schedule.create" is within an "async createScheduleWithAvailabilities" which is an async function, which means it always returns a promise, so it's already up to the caller of "createScheduleWithAvailabilities" whether or not to await it.

Reply:
So its essentially the same thing